storage/backup-restore/src/lib.rs

```diff
@@ -1,7 +1,68 @@
 // Copyright (c) The Libra Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-#[allow(dead_code)]
-mod adapter;
+pub mod adapter;
+
+use crate::adapter::Adapter;
+use anyhow::Result;
+use futures::{stream, StreamExt};
+use libra_types::transaction::Version;
+use storage_client::{StorageRead, StorageReadServiceClient};
 
 type FileHandle = String;
+
+pub async fn backup_account_state(
+    client: &StorageReadServiceClient,
+    version: Version,
+    adapter: &impl Adapter,
+    max_chunk_size: usize,
+) -> Result<Vec<FileHandle>> {
+    let account_stream = client.backup_account_state_async(version)?;
+
+    let ret = backup_data(
+        account_stream.map(|res| {
+            let resp = res.expect("Bad account data.");
+            let mut bytes = vec![];
+            bytes.extend(lcs::to_bytes(&resp.account_key).expect("Bad account key."));
+            bytes.extend(lcs::to_bytes(&resp.account_state_blob).expect("Bad account blob."));
+            bytes
```

Comment:
I think if you merge this with backup_data, you'll have the account_key, account_state_blob available and it will be much easier to ask for the proof.

Reply:
Yeah I'd like to land https://github.com/libra/libra/pull/2090 first otherwise we don't have the API at all.